lib/pure/typetraits.nim

```diff
@@ -57,6 +57,16 @@ proc supportsCopyMem*(t: typedesc): bool {.magic: "TypeTrait".}
   ## This trait returns true iff the type ``t`` is safe to use for
   ## `copyMem`:idx:. Other languages name a type like these `blob`:idx:.
 
+proc `==`*(t1, t2: typedesc): bool =
+  ## Returns wheter ``t1``, ``t2`` are the same type; this is different
+  ## from ``t1 is t2`` since the latter supports concepts & inheritance.
+  runnableExamples:
+    type T=int
+    doAssert T == int
+    doAssert int == T
+    doAssert: int != float
```

Comment:
Please be consistent in your style, no need for this random ``:``.

Reply:
it's needed to avoid https://github.com/nim-lang/Nim/issues/8300
otherwise you get: `Error: invalid indentation`

Reply:
I doubt it. ``!=`` is not an unary keyword operator.

Reply:
you can try for yourself with `git fetch origin pull/8634/head:pull_8634 && git checkout pull_8634` ...

Reply:
This compiles:

```nim

import typetraits

type T = int

template `==`(x, y: typedesc): bool = (x is y) and (y is x)

doAssert T == int
doAssert int == T
doAssert int != float
```

Reply:
ok, this must've been fixed since that was reported